perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1558,6 +1562,21 @@ public Application rejectApplication(PerunSession sess, int appId, String reason
 			log.info("Application {} marked as REJECTED.", appId);
 
 			deleteApplicationReservedLogins(sess, app);
+			// reject all of user's other group applications to the VO
+			if (app.getGroup() == null && app.getType().equals(AppType.INITIAL)) {
+				// create dummy principal and session using users information to pass it into getOpenApplicationsForUserInVo
+				PerunPrincipal userPrincipal = new PerunPrincipal(app.getCreatedBy(), "", "", app.getUser());
+				userPrincipal.setExtSourceName(app.getExtSourceName());
+				userPrincipal.setExtSourceType(app.getExtSourceType());
+				userPrincipal.setAdditionalInformations(BeansUtils.stringToMapOfAttributes(app.getFedInfo()));
+				PerunSession helperSess = new PerunSessionImpl(sess.getPerun(), userPrincipal, sess.getPerunClient());
+				List<Application> otherApps = registrarManager.getOpenApplicationsForUserInVo(helperSess, app.getVo());
+				for (Application otherApp : otherApps) {
+					if (otherApp.getGroup() != null) {
```

Comment:
Is this check necessary?

Reply:
Now that I'm aware only one app per user in vo, nope